app/src/main/java/ch/epfl/cs311/wanderwave/model/data/ProfileTrackAssociation.kt

```diff
@@ -9,9 +9,24 @@ package ch.epfl.cs311.wanderwave.model.data
  * @since 2.0
  * @last update 2.0
  */
-data class ProfileTrackAssociation(val profile: Profile? = null, val track: Track) {
+data class ProfileTrackAssociation(
+    val profile: Profile? = null,
+    val track: Track,
+    val likers: List<Profile> = emptyList(),
+    val likes: Int = 0
+) {
 
   fun toMap(): Map<String, Any?> {
     return hashMapOf("profile" to profile?.toMap(), "track" to track.toMap())
   }
+
+  fun likeTrack(profile: Profile): ProfileTrackAssociation {
+    this.profile?.numberOfLikes = this.profile?.numberOfLikes!! + 1
```

Comment:
This seems like a bad null assert, Can you add a test before that test if the profile is not null, 
something like 
`this.profile?.let{...}
?: // case if it's null`

Reply:
And maybe add a safety check that the profile is not already in the likers.